src/availability.rs

```diff
@@ -325,6 +325,44 @@ where
                     .unwrap())
             }
             .boxed()
+        })?
+        .get("get_block_summary", move |req, state| {
+            async move {
+                let id: usize = req.integer_param("height")?;
+
+                state
+                    .get_block(BlockId::Number(id))
+                    .await
+                    .with_timeout(timeout)
+                    .await
+                    .context(FetchBlockSnafu {
+                        resource: id.to_string(),
+                    })
+                    .map(|block| Into::<BlockSummaryQueryData<Types>>::into(block))
```

Comment:
Nit: slightly more idiomatic to say `BlockSummaryQueryData::from(block)`. You might even be able to do `block.into()`, but maybe the compiler can't infer the type there